src/defaultsettings.cpp

```diff
@@ -26,6 +26,68 @@ with this program; if not, write to the Free Software Foundation, Inc.,
 #include "mapgen/mapgen.h" // Mapgen::setDefaultSettings
 #include "util/string.h"
 
+
+
+bool detect_touch()
+{
+#if ENABLE_TOUCH  // Android & special builds
+	return true;
+#else
+	std::string chassis_type;
+#if defined(__linux__) || defined(__ANDROID__) || defined(__FreeBSD__) || defined(__NetBSD__) || defined(__DragonFly__)
+	// device-tree platforms (non-X86)
+	std::ifstream dtb_file("/proc/device-tree/chassis-type");
+	if (dtb_file.is_open()) {
+		std::getline(dtb_file, chassis_type);
+		chassis_type.pop_back();
+
+		if (chassis_type == "convertible" || // user has to set it by hand anyway
+		    chassis_type == "desktop" ||
+		    chassis_type == "laptop" ||
+		    chassis_type == "server" ||
+		    chassis_type == "embedded")
+			return false;
+
+		if (chassis_type == "tablet" ||
+		    chassis_type == "handset" ||
+		    chassis_type == "watch")
+			return true;
+
+		return false;
+	}
+	// SMBIOS (x86)
+	std::ifstream dmi_file("/sys/class/dmi/id/chassis_type");
+	if (dmi_file.is_open()) {
+		std::getline(dmi_file, chassis_type);
+
+		if (chassis_type == "11" /* Handheld */ ||
+		    chassis_type == "30" /* Tablet */)
+			return true;
+
+		return false;
+	}
+
+	// ACPI-based platforms
+	std::ifstream acpi_file("/sys/firmware/acpi/pm_profile");
+	if (acpi_file.is_open()) {
+		std::getline(acpi_file, chassis_type);
+
+		if (chassis_type == "8" /* Tablet */)
+			return true;
+
+		return false;
+	}
+#elif defined(_WIN32)
+	if (GetSystemMetrics(SM_TABLETPC))
+		return true;
```

Comment:
>  Nonzero if the current operating system is the Windows XP Tablet PC edition or if the current operating system is Windows Vista or Windows 7 and the Tablet PC Input service is started; otherwise, 0. The SM_DIGITIZER setting indicates the type of digitizer input supported by a device running Windows 7 or Windows Server 2008 R2. For more information, see Remarks.  
> https://learn.microsoft.com/en-us/windows/win32/api/winuser/nf-winuser-getsystemmetrics

looks like this doesn't work on recent Windows versions?

Reply:
Switched to the `SM_DIGITIZER`, I'll check after 10.3. on my laptop w/ Win+touchscreen.